src/helpers/compareMaps.ts

```diff
@@ -0,0 +1,15 @@
+export function compareMaps(map1 : Map<any,any>, map2 : Map<any,any>) {
+    let testVal;
+    if (map1.size !== map2.size) {
+        return false;
+    }
+    for (let [key, val] of map1) {
+        testVal = map2.get(key);
+        // in cases of an undefined value, make sure the key
+        // actually exists on the object so there are no false positives
+        if (testVal !== val || (testVal === undefined && !map2.has(key))) {
+            return false;
+        }
+    }
+    return true;
+}
```

Comment:
Ahaha this is very python'y, there might be a better way to do this for us JS/Reactors ;)

Reply:
I'm not sure, is there an existing package for this? Or are you talking about inline checking? Would appreciate a hint! :)

Reply:
Sorry, JS / React developers ahahah

Reply:
Have you heard of a reduce function? But before I send you off on a tangent can you just explain the purpose of the function / what it is doing quickly for me :)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

Reply:
Not by that name, but I have used it before; it runs a function iteratively over elements in the array and "reduces" the elements into a single value? 

I am using the `compareMaps()` function to check for deep equality between two maps, to be honest I should have called it `areMapsDeepEqual()`

The function is checking
1. whether the maps are of the same length
2. whether the KV pairs in each map matches the other

If these two conditions are fulfilled then the function returns true; I used this [solution](https://stackoverflow.com/questions/35948335/how-can-i-check-if-two-map-objects-are-equal)